server/proxy/pf_context.c

```diff
@@ -79,3 +79,32 @@ rdpContext* p_client_context_create(rdpSettings* clientSettings,
 	settings->RedirectClipboard = FALSE;
 	return context;
 }
+
+proxyData* pf_context_proxy_data_new()
+{
+	proxyData* pdata = malloc(sizeof(proxyData));
+
+	if (pdata == NULL)
+	{
+		return NULL;
+	}
+
+	pdata->info = malloc(sizeof(connectionInfo));
```

Comment:
i would add a wrapper for `connectionInfo` stuff, like `connectionInfo_new()` and `connectionInfo_free()`

Reply:
can be kept inside `pf_context`

Reply:
I don't like the idea of exporting connectionInfo_new. it's a part of proxyData and is set during different parts of the connection, therefore there is no need for connectionInfo_new function that will just malloc the struct. I've added a connectionInfo_free function that frees its members